app/javascript/components/timeline-options/timeline-helper.js

```diff
@@ -0,0 +1,115 @@
+export const timelineUiOptions = (title) => ({
+  title,
+  axes: {
+    bottom: {
+      title: __('Date'),
+      mapsTo: 'date',
+      scaleType: 'time',
+    },
+    left: {
+      title: __('# of Events'),
+      mapsTo: 'value',
+      scaleType: 'linear',
+    },
+  },
+  legend: {
+    clickable: true,
+  },
+  tooltip: {
+    totalLabel: __('Total Events'),
+  },
+  points: {
+    radius: 3,
+    fillOpacity: 1,
+    filled: true,
+    enabled: true,
+  },
+  zoomBar: {
+    top: {
+      className: 'zoom-bar',
+      enabled: true,
+      type: 'graph_view',
+    },
+  },
+  height: '400px',
+});
+
+const smartAndOrStatements = (group, array) => {
+  const item = (array.length === 1) ? array[0] : `[${array.toString()}]`;
+  return `&filter[]=${group}=${item}`;
+};
+
+/** Function to build the URL to get all events that fall under the user selected parameters. */
+export const buildUrl = (values, pageValues) => {
+  // TODO: Different timeline show different data, ensure all necesary data is pulled
+  let url = `/api/event_streams?limit=5000&offset=0&expand=resources`;
+  url += `&attributes=group,group_level,group_name,id,event_type,ems_id,type,timestamp,created_on,host.name,source,host_id`;
+
+  // Page / Record set values
+  url += `&filter[]=ems_id=${pageValues.ems_id}`;
+  url += pageValues.host_id ? `&filter[]=host_id=${pageValues.host_id}` : ``;
+
+  // User set values
+  url += `&filter[]=type=${values.type}`;
+  url += smartAndOrStatements('group', values.group);
+  url += smartAndOrStatements('group_level', values.group_level);
+
+  if (values.start_date) {
+    url += `&filter[]=timestamp%3E${values.start_date[0].toISOString()}`;
+  }
+  if (values.end_date) {
+    url += `&filter[]=timestamp%3C${values.end_date[0].toISOString()}`;
```

Comment:
These values are assumed to be there.  Maybe these should be smart defaults or perhaps we raise an error if we're unsure.